shardingsphere-distsql/shardingsphere-distsql-parser/src/main/java/org/apache/shardingsphere/distsql/parser/core/advanced/AdvancedDistSQLStatementParserEngine.java

```diff
@@ -40,12 +40,14 @@ public SQLStatement parse(final String sql) {
         ASTNode astNode = parseToASTNode(sql);
         return getSQLStatement(sql, (ParseASTNode) astNode);
     }
-    
+
     private ASTNode parseToASTNode(final String sql) {
         try {
             return SQLParserFactory.newInstance(sql, AdvancedDistSQLLexer.class, AdvancedDistSQLParser.class).parse();
-        } catch (final ParseCancellationException | SQLParsingException ignored) {
-            throw new SQLParsingException("You have an error in your SQL syntax.");
+        } catch (final ParseCancellationException ignored) {
+            throw new SQLParsingException("You have an error in your SQL syntax: " + sql);
+        } catch (final SQLParsingException sqlParsingException) {
+            throw sqlParsingException;
```

Comment:
I couldn't test this line from the tests. Any invalid sql I generate throws ParseCancellationException and therefore line 48 is being tested. Any advice on how I can test this line @terrymanu?

Reply:
Can you try to input an error SQL from JDBC and Proxy, and parse the exception here.
I want to make sure the exception thrown on the right way.